test/src/waittask.spec.ts

```diff
@@ -52,55 +52,18 @@ describe('waittask specs', function () {
     });
     it('should poll on interval', async () => {
       const {page} = getTestState();
-
-      let success = false;
       const startTime = Date.now();
       const polling = 100;
-      const watchdog = page
-        .waitForFunction(
-          () => {
-            return (globalThis as any).__FOO === 'hit';
-          },
-          {
-            polling,
-          }
-        )
-        .then(() => {
-          return (success = true);
-        });
-      await page.evaluate(() => {
-        return ((globalThis as any).__FOO = 'hit');
-      });
-      expect(success).toBe(false);
+      const watchdog = page.waitForFunction(
+        () => {
+          return (globalThis as any).__FOO === 'hit';
+        },
+        {polling}
+      );
       await page.evaluate(() => {
-        return document.body.appendChild(document.createElement('div'));
-      });
-      await watchdog;
-      expect(Date.now() - startTime).not.toBeLessThan(polling / 2);
-    });
-    it('should poll on interval async', async () => {
-      const {page} = getTestState();
-      let success = false;
-      const startTime = Date.now();
-      const polling = 100;
-      const watchdog = page
-        .waitForFunction(
-          async () => {
-            return (globalThis as any).__FOO === 'hit';
-          },
-          {
-            polling,
-          }
-        )
-        .then(() => {
-          return (success = true);
-        });
-      await page.evaluate(async () => {
-        return ((globalThis as any).__FOO = 'hit');
-      });
-      expect(success).toBe(false);
-      await page.evaluate(async () => {
-        return document.body.appendChild(document.createElement('div'));
+        setTimeout(() => {
```

Comment:
why is there a need for the timeout now?

Reply:
This timeout is still needed because it's not guaranteed the poller started. We wait 50 milliseconds because that's half the polling duration; by that time, the poller will for sure have started.

Reply:
Could we use setImmediate or setTimeout(0) too? does the poller start on the next tick? or can we have an event that the poller started?

Reply:
I've tried setTimeout(0) and it's flaky. Since the test requires at least 50ms anyways, this is fine.